Remove need for rocket_cors in local dev

ELD opened the pull request:
Swapped server routes to /api and proxied requests that are unknown to
the frontend app to the backend server to be serviced by Rocket

This is a reasonably simple PR, but it gets rid of the need for `rocket_cors` in development. Production will have a similar setup where `/api` routes go to the Rocket executable while all others go to static assets or the frontend JS bundle/app.

#### Before you submit this PR for review, make sure the following tasks are complete.

- [x] Added relevant documentation.
- [x] Updated unit tests.
- [x] Made sure all tests are passing with `yarn test`
- [x] Ensured all code meets the [standard JS](https://standardjs.com) style guide.
- [x] Removed debug code.
- [x] Included screenshot if applicable.
- [x] Squashed changes into a single commit.
- [x] If adding new routes to the frontend application, specify them below for testing

    ```
    git rebase -i $(git merge-base HEAD master)
    git push origin NAME_OF_BRANCH --force-with-lease
    ```

MichaelJamesParsons commented on bin/start:
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# Ensure Foreman is installed
+foreman --help &> /dev/null || gem install foreman
```
`gem install` assumes ruby is present on the OS. This should also be documented in the app's readme so devs know to install it ahead of time.

ELD replied:
Fair point. I enjoy using `foreman` because it just works with a Procfile and lets you spin up both the frontend and server in the same process.

MichaelJamesParsons commented on titan-web-client/package.json:
```diff
@@ -132,5 +132,6 @@
     ]
-  }
+  },
+  "proxy": "http://[::1]:8000"
```
The ipv6 address isn't playing well with my linux environment. I'll have to look into it. Is there a reason why you chose this over ipv4?

ELD replied:
IPv4 doesn't play well on the Mac as far as I can tell. I'll test it again and see if we can just bind to `127.0.0.1`.

MichaelJamesParsons commented on bin/start:
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
This script should be documented so new devs are aware of it.

MichaelJamesParsons commented on titan-server/src/bin/main.rs:
```diff
@@ -8,30 +8,27 @@ extern crate diesel;
     rocket::ignite()
         .attach(db::TitanPrimary::fairing())
         .attach(db::UnksoMainForums::fairing())
-        .attach(cors_fairing::cors())
```
If we're no longer using `rocket_cors`, it should be removed from the apps dependencies in `Cargo.toml`.

ELD replied:
Yeah, I wasn't sure if I had completely removed all references to it. In fact, there are some in the library code that I need to deal with.

ELD commented:
@MichaelJamesParsons Addressed the above points.

Did the following:
- Added note about `bin/start` to the main README.md
- Switched from Ruby `foreman` to Rust `fors` (is automatically installed if not available via `cargo install`)
- Removed remaining references to `rocket_cors`

MichaelJamesParsons commented:
@ELD Looks good. Did you get a chance to test running the app on `127.0.01`?

ELD commented:
@MichaelJamesParsons Yeah, it doesn't work on Mac, sadly. Not sure why, but it gets an ECONNREFUSED error.

MichaelJamesParsons commented:
I get the same for `[::1]` on linux. I was able to get ipv6 to work with a php/apache2 app, so it seems to be specific to titan.

We should look into this further before merging the PR.

ELD commented:
@MichaelJamesParsons This should be fixed. Rocket binds to `[::1]` in the absence of an explicit address. I had to add `address = 127.0.0.1` to the `Rocket.toml` to get it to bind to the IPv4 version of `localhost`. It should work on both Mac and Linux now.

It's weird that Linux still defaults to IPv4 and macOS defaults to IPv6. At least the Rust libraries do.

MichaelJamesParsons commented:
I'll test it out tonight. The branch conflicts must be resolved before merging.

ELD commented:
I’m aware. As soon as you check it out, approve the PR and I’ll rebase and merge it against master (and resolve whatever conflicts might exist).
